crates/solidity/outputs/cargo/crate/src/extensions/utils/version/mod.rs

```diff
@@ -0,0 +1,458 @@
+
+                ComparatorSet::bounds(lower, upper)
+            }
+            None => ComparatorSet::wild(),
+        }
+    }
+
+    /// Create a `ComparatorSet` which represents the range of versions described by `[lower_version] - [upper_version]`.
+    fn hyphen_range(
+        lower_version: &PartialVersion,
+        upper_version: &PartialVersion,
+    ) -> ComparatorSet {
+        let lower = Comparator {
+            version: lower_version.lower_bound(),
+            op: Operator::GtEq,
+        };
+
+        let upper = match upper_version.major {
+            VersionPart::Specified(major) => match upper_version.minor {
+                VersionPart::Specified(minor) => match upper_version.patch {
+                    VersionPart::Specified(patch) => Comparator {
+                        version: Version::new(major, minor, patch),
+                        op: Operator::LtEq,
+                    },
+                    _ => Comparator {
+                        version: Version::new(major, minor + 1, 0),
+                        op: Operator::Lt,
+                    },
+                },
+                _ => Comparator {
+                    version: Version::new(major + 1, 0, 0),
+                    op: Operator::Lt,
+                },
+            },
+            _ => Comparator::wild(),
+        };
+
+        ComparatorSet::bounds(lower, upper)
+    }
+
+    /// Create a `ComparatorSet` that represents the range described by `[partial]`.
+    /// A partially specified version represents a version range, sometimes called an x-range.
+    /// `1.x == >=1.0.0 <2.0.0`
+    /// `0.4.x == >=0.4.0 <0.5.0`
+    /// `2.1 == >=2.1.0 < 2.2.0`
+    fn partial_range(partial: &PartialVersion) -> ComparatorSet {
+        if partial.major.is_wild() {
+            return ComparatorSet::single(Comparator::default());
+        }
+
+        if partial.is_complete() {
+            return ComparatorSet::single(Comparator {
+                version: partial.lower_bound(),
+                op: Operator::Eq,
+            });
+        }
+
+        let lower = Comparator {
+            version: partial.lower_bound(),
+            op: Operator::GtEq,
+        };
+
+        let upper = match partial.major {
+            VersionPart::Specified(major) => match partial.minor {
+                VersionPart::Specified(minor) => match partial.patch {
+                    // Handled above by checking partial.is_complete()
+                    VersionPart::Specified(_) => unreachable!(),
+                    _ => Comparator {
+                        version: Version::new(major, minor + 1, 0),
+                        op: Operator::Lt,
+                    },
+                },
+                _ => Comparator {
+                    version: Version::new(major + 1, 0, 0),
+                    op: Operator::Lt,
+                },
+            },
+            // Handled above by checking partial.major.is_wild()
+            _ => unreachable!(),
+        };
```

Comment:
This could be written as:
```
let upper = Comparator {
  version: partial.tilde_upper_bound(),
  op: Operator::Lt,
};
```
Or am I missing something?